distributed/shuffle/_shuffle.py

```diff
@@ -82,13 +85,17 @@ def shuffle_unpack(
         )
     except Reschedule as e:
         raise e
+    except ShuffleClosedError:
+        raise Reschedule()
     except Exception as e:
         raise RuntimeError(f"shuffle_unpack failed during shuffle {id}") from e
 
 
 def shuffle_barrier(id: ShuffleId, run_ids: list[int]) -> int:
     try:
         return _get_worker_plugin().barrier(id, run_ids)
+    except ShuffleClosedError:
+        raise Reschedule()
```

Comment:
This lack of coverage make me nervous. I think around the barrier there are various races we should test.

Reply:
I haven't been able to come up with a scenario where this would be triggered, so I've removed it for now. If this ever pops up for somebody, I hope they'll send a bug report our way.